scripts/tests/yaml/runner.py

```diff
@@ -0,0 +1,294 @@
+                self.params.append(param)
+
+        ctx.custom_options = custom_options
+
+
+CONTEXT_SETTINGS = dict(
+    default_map={
+        'chiptool': {
+            'adapter': 'matter_chip_tool_adapter.adapter',
+            'server_name': 'chip-tool',
+            'server_arguments': 'interactive server',
+        },
+        'app1': {
+            'test_directory': 'examples/placeholder/linux/apps/app1',
+            'adapter': 'matter_placeholder_adapter.adapter',
+            'server_name': 'chip-app1',
+            'server_arguments': '--interactive',
+        },
+        'app2': {
+            'test_directory': 'examples/placeholder/linux/apps/app2',
+            'adapter': 'matter_placeholder_adapter.adapter',
+            'server_name': 'chip-app2',
+            'server_arguments': '--interactive',
+        },
+    },
+    max_content_width=120,
+)
+
+
+@click.group(cls=YamlTestParserGroup, context_settings=CONTEXT_SETTINGS)
+@click.argument('test_name')
+@test_parser_options
+@click.pass_context
+def runner_base(ctx, test_directory: str, test_name: str, configuration_name: str, pics: str, specifications_paths: str, stop_on_error: bool, **kwargs):
+    specifications = SpecDefinitionsFromPaths(specifications_paths.split(','), get_default_pseudo_clusters())
+    tests_finder = TestsFinder(test_directory, configuration_name)
+
+    parser_config = TestParserConfig(pics, specifications, kwargs)
+    parser_builder_config = TestParserBuilderConfig(tests_finder.get(test_name), parser_config, hooks=TestParserLogger())
+    parser_builder_config.options.stop_on_error = stop_on_error
+    while ctx:
+        ctx.obj = parser_builder_config
+        ctx = ctx.parent
+
+
+@runner_base.command()
+@pass_parser_builder_config
+def parse(parser_builder_config: TestParserBuilderConfig):
+    """Parse the test suite."""
+    runner_config = None
+
+    runner = TestRunner()
+    return runner.run(parser_builder_config, runner_config)
+
+
+@runner_base.command()
+@pass_parser_builder_config
+def dry_run(parser_builder_config: TestParserBuilderConfig):
+    """Simulate a run of the test suite."""
+    runner_config = TestRunnerConfig(hooks=TestRunnerLogger())
+
+    runner = TestRunner()
+    return runner.run(parser_builder_config, runner_config)
+
+
+@runner_base.command()
+@test_runner_options
+@pass_parser_builder_config
+def run(parser_builder_config: TestParserBuilderConfig, adapter: str, stop_on_error: bool, stop_on_warning: bool, stop_at_number: int, show_adapter_logs: bool, show_adapter_logs_on_error: bool):
+    """Run the test suite."""
+    adapter = __import__(adapter, fromlist=[None]).Adapter(parser_builder_config.parser_config.definitions)
+    runner_options = TestRunnerOptions(stop_on_error, stop_on_warning, stop_at_number)
+    runner_hooks = TestRunnerLogger(show_adapter_logs, show_adapter_logs_on_error)
+    runner_config = TestRunnerConfig(adapter, get_default_pseudo_clusters(), runner_options, runner_hooks)
```

Comment:
Maybe add a TODO that maybe we don't want get_default_pseudo_clusters here all the time

Reply:
I have updated the code with an option to disable the pseudo clusters.